armory/scenarios/scenario.py

```diff
@@ -220,16 +220,25 @@ def load_metrics(self):
             computational_resource_dict=metrics_logger.computational_resource_dict,
         )
 
-        export_samples = self.config["scenario"].get("export_samples")
-        if export_samples is not None and export_samples > 0:
-            sample_exporter = SampleExporter(
-                self.scenario_output_dir, self.test_dataset.context, export_samples
+        self.metrics_logger = metrics_logger
+
+    def load_sample_exporter(self):
+        if self.config["scenario"].get("export_samples") is not None:
+            log.warning(
+                "The export_samples field was deprecated in Armory 0.15.0. Please use export_batches instead."
             )
-        else:
-            sample_exporter = None
 
-        self.metrics_logger = metrics_logger
-        self.sample_exporter = sample_exporter
+        num_export_batches = self.config["scenario"].get("export_batches", 0)
+        if num_export_batches is True:
+            num_export_batches = len(self.test_dataset)
+        self.num_export_batches = num_export_batches
+        self.sample_exporter = self._load_sample_exporter()
+
+    @abc.abstractmethod
+    def _load_sample_exporter(self):
+        raise NotImplementedError(
+            f"_load_sample_exporter() method should be defined for scenario {self.__class__}"
+        )
```

Comment:
I'm not sure that we want to enforce this method for Scenario, as we haven't enforced other methods. Instead, I would probably just have a runtime error like this:
```
    def _load_sample_exporter(self):
        raise NotImplementedError(
            f"_load_sample_exporter() is not implemented for scenario {self.__class__}"
        )
```